sklearn/tests/test_docstring_parameters.py

```diff
@@ -228,9 +228,19 @@ def test_fit_docstring_attributes(name, Estimator):
                'StackingRegressor', 'TfidfVectorizer', 'VotingClassifier',
                'VotingRegressor', 'SequentialFeatureSelector',
                'HalvingGridSearchCV', 'HalvingRandomSearchCV'}
+
+    # TODO remove in 1.1 or 1.2 when the deprecated attributes are
+    # removed
+    HAS_DEPRECATED_ATTRS = {'Birch', 'CCA', 'CategoricalNB', 'MiniBatchKMeans',
+                            'PLSCanonical', 'PLSRegression', 'PLSSVD',
+                            'SpectralCoclustering'}
+
     if Estimator.__name__ in IGNORED or Estimator.__name__.startswith('_'):
         pytest.skip("Estimator cannot be fit easily to test fit attributes")
 
+    if Estimator.__name__ in HAS_DEPRECATED_ATTRS:
+        pytest.skip("Estimator has undocumented deprecated attributes")
```

Comment:
Turning off the whole test for an estimator because it has deprecated properties means we will not catch future attribute docstring errors. In the property finding code can we filter out deprecated properties? For example:

```python
def _get_all_fitted_attributes(estimator):
    "Get all the fitted attributes of an estimator including properties"
    # attributes
    fit_attr = list(estimator.__dict__.keys())

    # properties
    with warnings.catch_warnings():
        warnings.filterwarnings("error", category=FutureWarning)

        for name in dir(estimator.__class__):
            obj = getattr(estimator.__class__, name)
            if not isinstance(obj, property):
                continue

            # ignore properties that raises an AttributeError and deprecated
            # properties
            try:
                getattr(estimator, name)
            except (AttributeError, FutureWarning):
                continue
            fit_attr.append(name)

    return [k for k in fit_attr if k.endswith('_') and not k.startswith('_')]

```

Reply:
Your solution is better, thanks. And it actually revealed that I excluded an estimator by mistake, the property was not deprecated.